src/Hng.Application/Features/Roles/Handler/CreateRoleCommandHandler.cs

```diff
@@ -0,0 +1,72 @@
+﻿using Hng.Application.Features.Roles.Command;
+using Hng.Application.Features.Roles.Dto;
+using Hng.Domain.Entities;
+using Hng.Infrastructure.Repository.Interface;
+using MediatR;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Hng.Application.Features.Roles.Handler
+{
+    public class CreateRoleCommandHandler : IRequestHandler<CreateRoleCommand, CreateRoleResponseDto>
+    {
+        private readonly IRepository<Organization> _organizationRepository;
+        private readonly IRepository<Role> _roleRepository;
+
+        public CreateRoleCommandHandler(IRepository<Organization> organizationRepository, IRepository<Role> roleRepository)
+        {
+            _organizationRepository = organizationRepository;
+            _roleRepository = roleRepository;
+        }
+
+        public async Task<CreateRoleResponseDto> Handle(CreateRoleCommand request, CancellationToken cancellationToken)
+        {
+            var organization = await _organizationRepository.GetAsync(request.OrganizationId);
+            if (organization == null)
```

Comment:
Are you creating role for organization or role for user?

Reply:
for a user under an organisation

each user can have different roles under different organisation